From ebiederm@xmission.com  Wed Jul 21 13:33:29 2010
To: Greg KH <gregkh@suse.de>
Cc: Andrew Morton <akpm@linux-foundation.org>, Greg KH <greg@kroah.com>,
        "Rafael J. Wysocki" <rjw@sisk.pl>,
        "Maciej W. Rozycki" <macro@linux-mips.org>,
        Kay Sievers <kay.sievers@vrfy.org>,
        Johannes Berg <johannes@sipsolutions.net>
Subject: sysfs: sysfs_delete_link handle symlinks from untagged to tagged directories.
From: ebiederm@xmission.com (Eric W. Biederman)
Date: Tue, 20 Jul 2010 22:10:58 -0700
Message-ID: <m1vd894dy5.fsf_-_@fess.ebiederm.org>


This happens for network devices when SYSFS_DEPRECATED is enabled.

Signed-off-by: Eric W. Biederman <ebiederm@xmission.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/sysfs/symlink.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/sysfs/symlink.c
+++ b/fs/sysfs/symlink.c
@@ -135,7 +135,7 @@ void sysfs_delete_link(struct kobject *k
 {
 	const void *ns = NULL;
 	spin_lock(&sysfs_assoc_lock);
-	if (targ->sd)
+	if (targ->sd && sysfs_ns_type(kobj->sd))
 		ns = targ->sd->s_ns;
 	spin_unlock(&sysfs_assoc_lock);
 	sysfs_hash_and_remove(kobj->sd, ns, name);
